index.js

```diff
@@ -104,51 +77,279 @@ module.exports = {
 
   errors: errors,
 
-  DEFAULT_HOSTED_NODES: HOSTED_NODES.slice(),
-
-  DEFAULT_HOSTED_WEBSOCKET: HOSTED_WEBSOCKET,
-
-  nodes: {
-    hosted: HOSTED_NODES.slice(),
-    local: null
-  },
-
-  requests: 1,
+  requests: null,
 
   // Hook for transaction callbacks
   txRelay: null,
 
   // Do not call txRelay for these methods
-  excludedFromTxRelay: {},
+  excludedFromTxRelay: null,
 
-  txs: {},
+  txs: null,
 
-  rawTxs: {},
+  rawTxs: null,
 
-  rawTxMaxNonce: -1,
+  rawTxMaxNonce: null,
 
   block: null,
 
   networkID: null,
 
-  notifications: {},
+  notifications: null,
+
+  gasPrice: null,
+
+  configuration: null,
+
+  internalState: null,
+
+  /**
+   * Initiates a connection to Ethereum.  This must be called before any other methods are called.
+   *
+   * @typedef configuration
+   * @type {object}
+   * @property {?string[]} httpAddresses
+   * @property {?string[]} wsAddresses
+   * @property {?string[]} ipcAddresses
+   * @property {?number} connectionTimeout
+   * @property {?function(Error):void} errorHandler - called when an otherwise unhandled asynchronous error occurs during the course of operation.
+   *
+   * @param {!configuration} configuration
+   * @param {!function(?Error):void} initialConnectCallback - if the error parameter is null then the connection was successful
+   * @returns {void}
+   */
+  connect: function (configuration, initialConnectCallback) {
+    this.resetState();
+
+    // overwrite configuration values with user config, throw away unused user config
+    for (var key in this.configuration) {
+      if (configuration[key] === undefined || configuration[key] === null) continue;
```

Comment:
I would like the `hasOwnProperty` check added, but it's not just for consistency.  It's a good idea to have the `hasOwnProperty` check because something else may have added properties to `Object`.  This shouldn't happen in the Augur UI, but it could result in unexpected behavior if ethrpc is used in some other environment where this is true.